packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel/method_channel_firestore.dart

```diff
@@ -57,13 +61,51 @@ class MethodChannelFirebaseFirestore extends FirebaseFirestorePlatform {
     );
   }
 
+  /// The [EventChannel] used for loadBundle
+  static EventChannel loadBundleChannel(String id) {
+    return EventChannel(
+      'plugins.flutter.io/firebase_firestore/loadBundle/$id',
+      const StandardMethodCodec(FirestoreMessageCodec()),
+    );
+  }
+
   /// Gets a [FirebaseFirestorePlatform] with specific arguments such as a different
   /// [FirebaseApp].
   @override
   FirebaseFirestorePlatform delegateFor({required FirebaseApp app}) {
     return MethodChannelFirebaseFirestore(app: app);
   }
 
+  @override
+  LoadBundleTaskPlatform loadBundle(Uint8List bundle) {
+    return MethodChannelLoadBundleTask(
+        task: channel.invokeMethod<String>('LoadBundle#snapshots'),
+        bundle: bundle,
+        firestore: this);
+  }
+
+  @override
+  Future<QuerySnapshotPlatform> namedQueryGet(String name,
+      {GetOptions options = const GetOptions()}) async {
+    try {
+      final Map<String, dynamic>? data = await MethodChannelFirebaseFirestore
+          .channel
+          .invokeMapMethod<String, dynamic>(
+        'Firestore#namedQueryGet',
+        <String, dynamic>{
+          'name': name,
+          'firestore': FirebaseFirestorePlatform.instance,
+          'source': getSourceString(options.source),
+        },
+      );
+
+      return MethodChannelQuerySnapshot(
+          FirebaseFirestorePlatform.instance, data!);
```

Comment:
Could we make the returned Map not nullable rather than force it to be none null?

Reply:
^^ for the `data!` property

Reply:
Not possible I'm afraid. That method expects a nullable response.

Reply:
Hardly. It's `invokeMapMethod` that returns a nullable map and it's a Flutter Api

Reply:
not sure if you're talking to me or Elliot?

Reply:
Oh was talking with Elliot. Your comment wasn't visible when I answered, sorry for the confusion!